src/private.rs

```diff
@@ -0,0 +1,118 @@
+use std::fmt;
+
+use super::{AbiEntry, AbiFunction, RootSchema, SCHEMA_VERSION};
+
+pub use schemars;
+use serde::{Deserialize, Serialize};
+
+/// Core ABI information, with schema version and identity hash.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct ChunkedAbiEntry {
+    /// Semver of the ABI schema format.
+    pub(crate) abi_schema_version: String,
+    #[serde(flatten)]
+    pub abi: AbiEntry,
+}
+
+impl ChunkedAbiEntry {
+    pub fn new(functions: Vec<AbiFunction>, root_schema: RootSchema) -> ChunkedAbiEntry {
+        Self {
+            abi_schema_version: SCHEMA_VERSION.to_string(),
+            abi: AbiEntry {
+                functions,
+                root_schema,
+            },
+        }
+    }
+
+    pub fn combine<I: IntoIterator<Item = ChunkedAbiEntry>>(
+        entries: I,
+    ) -> Result<ChunkedAbiEntry, AbiCombineError> {
+        let mut abi_schema_version = None;
+        let mut functions = Vec::<AbiFunction>::new();
```

Comment:
Noting that there is no defined ordering here. Is there any guarantees around how the AbiEntries are sent in as the iterator? Wondering if this should be something like a `BTreeSet` to be able to ensure no duplicates and that there is a consistent ordering? cc @itegulov wdyt?

Reply:
> Noting that there is no defined ordering here.

They're later sorted by name.

https://github.com/near/near-abi-rs/blob/d5a7bca3a490f5e367c5a539764eaad47dfcfbc8/src/private.rs#L66-L67

> Wondering if this should be something like a BTreeSet to be able to ensure no duplicates

Rust already helps us with this guarantee? You can't have multiple methods with the same name implemented for the same type.

Reply:
True, I suppose this is a reasonable enough assumption, that this method is called only for our specific use case correctly (just trying to think of how to make it impossible to misuse, so it isn't just silently accepting if there is a bug).

Ignore this point

Reply:
Yeah, we could (and should in the future) add some sanity checks to the end, but this should work for our specific use case